### DISCUSSION TITLE
Save csv file of aeroacustics plots

### DISCUSSION THREAD

**Discussion by AndreaRapi** [byuflowlab/FLOWUnsteady#154] [Q&A] [answered]
There is a way to save the FLOWUnsteady Tonal SPL spectrum results (picture in blue) as a csv file using FLOWUnsteady scripts? I tried to modify some but it is not possible.
![image](https://github.com/byuflowlab/FLOWUnsteady/assets/148783703/6b6bb855-4bbe-457f-9efa-45ec4a598789)

Many thanks,
Andrea

**Comment by MarcoPic99** [comment]
Hi! may I ask you who did you contact to get access to PSU-WOPWOP? Thank you.

**Comment by AndreaRapi** [comment]
Hi @MarcoPic99,

To run the aeroacoustics package you require a copy of PSU-WOPWOP (an FW-H code that was written at Penn State) which is not included by FLOWUnsteady. You can request a copy of the binary by contacting them directly.

Ciao,
Andrea

**Reply by nrnaveenkumr** [reply]
Hi @AndreaRapi , can you share the email to contact for binary request.

thanks in advance

**Comment by ilm71** [comment]
> There is a way to save the FLOWUnsteady Tonal SPL spectrum results (picture in blue) as a csv file using FLOWUnsteady scripts? I tried to modify some but it is not possible. ![image](https://private-user-images.githubusercontent.com/148783703/309227377-6b6bb855-4bbe-457f-9efa-45ec4a598789.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAzMjE2NzQsIm5iZiI6MTcxMDMyMTM3NCwicGF0aCI6Ii8xNDg3ODM3MDMvMzA5MjI3Mzc3LTZiNmJiODU1LTRiYmUtNDU3Zi05ZWZhLTQ1ZWM0YTU5ODc4OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxM1QwOTE2MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jYzgyY2Q2MDMwY2UxMmQ4MzFkMzJhZWMxMmRiNDc2ZjQ5YWY3NTI0MzU4NDJlOGViMTEwZjBlMDgyYTk3MGE0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.cArZfuiS4FPB3FCU9e8bEQl5aTk_IqXT5AYePm6JS7M)
> 
> Many thanks, Andrea

I would be interested in that function too

**Reply by AndreaRapi** [reply]
Hi @philm71,

Have you had the opportunity to work on improving this feature?

**Reply by ilm71** [reply]
Hi @AndreaRapi, unfortunately not  -have you made some progress?

**Reply by AndreaRapi** [reply]
Not yet but I'll try. Thanks for the info.

**Reply by ilm71** [reply]
Great, thanks!! Could you please share the information if you have some news regarding this topic?

**Comment by EdoAlvarezR** [comment, marked as answer]
All the acoustics data that was generated by PSU-WOPWOP and BPM is stored under the dictionaries `datasets_pww` and `datasets_bpm` after calling
```julia
noise.read_data(dataset_infos; datasets_pww=datasets_pww, datasets_bpm=datasets_bpm)
```
You can access those dictionaries to fetch frequencies, SPL, and OASPL directivity (and perhaps save it as CSV if that format is better suited for your needs).

I'd recommend taking a look at the plotting functions to see how to process the PSU-WOPWOP and BPM data that is stored in those dictionaries: [LINK1](https://github.com/byuflowlab/FLOWNoise/blob/master/src/FLOWNoise_postprocessing.jl#L564) [LINK2](https://github.com/byuflowlab/FLOWNoise/blob/master/src/FLOWNoise_postprocessing.jl#L239)
